src/main/scala/is/hail/expr/ir/IR.scala

```diff
@@ -0,0 +1,74 @@
+package is.hail.expr.ir
+
+import is.hail.asm4s.TypeInfo
+
+import scala.language.existentials
+
+sealed case class PrimitiveOp(x: Long)
```

Comment:
I'm concerned about the usability of this in the term rewriter for, say, simplifying arithmetic expressions.  What's the vision here?